### PR TITLE
MINOR Reduce some code complexity, update array syntax and injected SQLSelect etc

### PR DESCRIPTION
**Pull request by robbieaverill**
This is just reducing some code smell and complexity, updating some array definitions and switching SQLSelect etc to be injectable in some of the list classes

### CODE REVIEW

File: src/ORM/ArrayList.php [hunk 1]
```diff
@@ -601,9 +601,9 @@ public function filter()
      */
     public function filterAny()
     {
-        $keepUs = call_user_func_array(array($this, 'normaliseFilterArgs'), func_get_args());
+        $keepUs = call_user_func_array([$this, 'normaliseFilterArgs'], func_get_args());
```
Comment (kinglozzer) [inline src/ORM/ArrayList.php]: `$keepUs = $this->normaliseFilterArgs(...func_get_args());` ? :D

File: src/ORM/ArrayList.php [hunk 2]
```diff
@@ -636,18 +636,18 @@ protected function normaliseFilterArgs($column, $value = null)
             throw new InvalidArgumentException('filter takes one array or two arguments');
         }
 
-        if (count(func_get_args()) == 1 && !is_array(func_get_arg(0))) {
+        if (count(func_get_args()) === 1 && !is_array(func_get_arg(0))) {
```
Comment (kinglozzer) [inline src/ORM/ArrayList.php]: Could use `$args = func_get_args()` and remove repeated calls to `func_get_args()`/`func_get_arg(n)`?

File: src/ORM/ArrayList.php [hunk 3]
```diff
@@ -719,23 +719,23 @@ public function filterByCallback($callback)
     public function exclude()
     {
 
-        $removeUs = call_user_func_array(array($this, 'normaliseFilterArgs'), func_get_args());
+        $removeUs = call_user_func_array([$this, 'normaliseFilterArgs'], func_get_args());
```
Comment (kinglozzer) [inline src/ORM/ArrayList.php]: `$removeUs = $this->normaliseFilterArgs(...func_get_args());` ? :D

File: src/ORM/DataList.php [hunk 4]
```diff
@@ -126,23 +126,23 @@ public function alterDataQuery($callback)
...
-                }
-            } catch (Exception $e) {
-                $list->inAlterDataQueryCall = false;
-                throw $e;
-            }
+        $list = clone $this;
+        $list->inAlterDataQueryCall = true;
 
+        try {
+            $res = call_user_func($callback, $list->dataQuery, $list);
```
Comment (kinglozzer) [inline src/ORM/DataList.php]: `$res = $callback($list->dataQuery, $list);`

File: src/ORM/DataList.php [hunk 5]
```diff
@@ -126,23 +126,23 @@ public function alterDataQuery($callback)
...
-                }
-            } catch (Exception $e) {
-                $list->inAlterDataQueryCall = false;
-                throw $e;
-            }
+        $list = clone $this;
+        $list->inAlterDataQueryCall = true;
 
+        try {
+            $res = $callback($this->dataQuery, $list);
```
Comment (kinglozzer) [inline src/ORM/DataList.php]: I think the failures are because this should be `$list->dataQuery`

Comment (robbieaverill) [reply src/ORM/DataList.php]: Whoops!

**Comment by robbieaverill** [comment]
Thanks for reviewing @kinglozzer!

**Pull request merged**